src/Hazelcast.Net.Tests/Networking/RegressionWithRealNetworkTest.cs

```diff
@@ -0,0 +1,258 @@
+
+            await AssertEx.SucceedsEventually(() =>
+            {
+                var clientImpl = (HazelcastClient)client;
+                Assert.AreEqual(1, clientImpl.Cluster.Connections.Count);
+                Assert.AreEqual(1, clientImpl.Members.Count);
+            }, 10_000, 500);
+
+            await RcClient.ShutdownMemberAsync(customCluster.Id, member.Uuid);
+            await Task.Delay(client.Options.Heartbeat.PeriodMilliseconds);
+            member = await RcClient.StartMemberAsync(customCluster.Id);
+
+            await AssertEx.SucceedsEventually(async () =>
+            {
+                await map.RemoveAsync(1);
+                await map.PutAsync(1, 2);
+                Assert.Greater(countOfEvent, 0);
+            }, 20_000, 500);
+
+            #region TearDown
+            await client.DisposeAsync();
+            await RcClient.ShutdownClusterAsync(customCluster);
+            #endregion
+        }
+
+        [Test]
+        [TestCase(true, ReconnectMode.ReconnectAsync)]
+        [TestCase(true, ReconnectMode.ReconnectSync)]
+        [TestCase(false, ReconnectMode.ReconnectAsync)]
+        [TestCase(false, ReconnectMode.ReconnectSync)]
+        public async Task TestOperationsContinueWhenClientDisconnected(bool smartRouting, ReconnectMode reconnectMode)
+        {
+            #region SetUp
+            using var _ = HConsoleForTest();
+            var customCluster = await RcClient.CreateClusterAsync();
+            var member1 = await RcClient.StartMemberAsync(customCluster.Id);
+
+            var client = await CreateAndStartClientAsync(options =>
+            {
+                options.Networking.SmartRouting = smartRouting;
+                options.Networking.ConnectionRetry.ClusterConnectionTimeoutMilliseconds = int.MaxValue;
+                options.Networking.ReconnectMode = reconnectMode;
+                options.ClusterName = customCluster.Id;
+            });
+
+            var mapName = "myTestingMap";
+            var member2 = await RcClient.StartMemberAsync(customCluster.Id);
+            #endregion
+
+            await AssertEx.SucceedsEventually(() => Assert.AreEqual(2, client.Members.Count), 10_000, 500);
+
+            var map = await client.GetMapAsync<int, int>(mapName);
+
+            await RcClient.ShutdownMemberAsync(customCluster.Id, member1.Uuid);
+
+            await AssertEx.SucceedsEventually(() =>
+            {
+                Assert.AreEqual(1, client.Members.Count);
+                Assert.AreEqual(member2.Host, client.Members.First().Member.Address.Host);
+                Assert.AreEqual(member2.Port, client.Members.First().Member.Address.Port);
+
+            }, 20_000, 500);
+            //Java test here also toys with partitions to verify that data migrates between members - irrrelevant to us
+            //details: https://github.com/hazelcast/hazelcast-csharp-client/pull/614
```

Comment:
I don't see relevance here. That method checks partition synchronization between members. Am I missing something?

Reply:
Please match Java code otherwise send PR to Java to change it if not relevant. I think that it has a logic that uses a key owned by member2 which is not shutdown and yes, it may not even be migrating since it will be backup promoting to primary.

Reply:
Opened a ticket to discuss to change it as well on Java side not to block the PR. Still, I don't see relevance with the client tests, let's discuss it.